lib/core/traversalUtils.js

```diff
@@ -204,25 +204,67 @@ export const getParentSelectorClassesMap = (ast: gASTNode): classMapType => {
   return classesMap;
 };
 
-/*
-   mutates ast by removing instances of :global
+/**
+ * Mutates the AST by removing `:global` instances.
+ *
+ * For the AST structure:
+ * @see https://github.com/css/gonzales/blob/master/doc/AST.CSSP.en.md
  */
 export const eliminateGlobals = (ast: gASTNode) => {
-  ast.traverse((node, index, parent) => {
-    if (node.type === 'ruleset') {
-      if (
-        fp.compose(
-          fp.negate(fp.isEmpty),
-          fp.find({ type: 'ident', content: 'global' }),
-          fp.get('content'),
-          fp.find({ type: 'pseudoClass' }),
-          fp.get('content'),
-          fp.find({ type: 'selector' }),
-          fp.get('content'),
-        )(node)
-      ) {
-        parent.removeChild(index);
+  // Remove all :global/:global(...) in selectors
+  ast.traverseByType('selector', (selectorNode) => {
+    const selectorContent = selectorNode.content;
+    let hasGlobalWithNoArgs = false;
+    let i = 0;
+    let currNode = selectorContent[i];
+    while (currNode) {
+      if (currNode.is('pseudoClass')) {
+        // Remove all :global/:global(...) and trailing space
+        const identifierNode = currNode.content[0];
+        if (identifierNode && identifierNode.content === 'global') {
+          if (currNode.content.length === 1) hasGlobalWithNoArgs = true;
+          selectorNode.removeChild(i);
+          if (selectorContent[i] && selectorContent[i].is('space')) {
+            selectorNode.removeChild(i);
+          }
+        } else {
```

Comment:
You might consider ensuring that `removeChild(i)` can't be called multiple times. Seems like that would be problematic.
```suggestion
        if (identifierNode && identifierNode.content === 'global') {
          if (currNode.content.length === 1) hasGlobalWithNoArgs = true;
          selectorNode.removeChild(i);
        }  else if (selectorContent[i] && selectorContent[i].is('space')) {
           selectorNode.removeChild(i);
        } else {
```

Reply:
It's expected to be called multiple times: first time to remove `:global` and second time to remove the trailing `space`. So in the following case:
```CSS
// before
:global(globalClass) .localClass {...}
```
`:global(globalClass) ` will be removed.
```CSS
// after
.localClass {...}
```